packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/ReviewWorkflows.js

```diff
@@ -58,11 +55,15 @@ export function ReviewWorkflowsPage() {
               id: 'Settings.review-workflows.page.subtitle',
               defaultMessage: '{count, plural, one {# stage} other {# stages}}',
             },
-            { count: STAGES.length }
+            { count: workflow?.workflows?.stages?.length ?? 0 }
           )}
         />
         <ContentLayout>
-          <Stages stages={STAGES} />
+          {workflow?.workflows?.state === 'loading' ? (
```

Comment:
Because `useInjectReducer` doesn't run before `useSelector` `workflow` might be undefined. We could fix this two ways I think:

- return initialState as part of useSelector.
- don't render until `workflow` is defined, which is probably cleaner

Reply:
Do you mean not use redux to store the state? I've put it there because I think it makes sense if redux manages the whole lifecycle and it follows the RTK pattern. In the long-term I guess this could be abstracted into it's own hook/ helper to keep react-query in sync with redux in terms of states.

As for the loading: Ok - I'll make this an early return 👍🏼